src/components/v2/appDetails/k8Resource/nodeType/PodHeader.component.tsx

```diff
@@ -1,83 +1,91 @@
+    const pods: Array<iNode> = IndexStore.getiNodesByKind('pod')
+    const newPodStats = { running: 0, all: 0 }
+    const oldPodStats = { running: 0, all: 0 }
+    const [newPods, setNewPods] = useState(newPodStats)
+    const [oldPods, setOldPods] = useState(oldPodStats)
 
     useEffect(() => {
-        if (podMetaData && podMetaData.length > 0) {
-            let _newPods = [];
-            let _oldPods = [];
-
-            podMetaData.forEach((pod) => {
-                if (pod.isNew) {
-                    _newPods.push(pod);
+        if (pods && podMetaData && podMetaData.length > 0) {
+            pods.forEach((pod) => {
+                const podStatusLower = getNodeStatus(pod)?.toLowerCase()
+                if (podMetaData.find((f) => f.name === pod.name)?.isNew) {
+                    newPodStats[podStatusLower] = (newPodStats[podStatusLower] || 0) + 1
+                    newPodStats['all'] += 1
                 } else {
-                    _oldPods.push(pod);
+                    oldPodStats[podStatusLower] = (oldPodStats[podStatusLower] || 0) + 1
+                    oldPodStats['all'] += 1
                 }
-            });
-            setNewPods(_newPods);
-            setOldPods(_oldPods);
-
-            setSelectedHealthyNewNodeCount(_newPods.length);
-            setSelectedHealthyOldNodeCount(_oldPods.length);
+            })
         }
-    }, [podMetaData?.length]);
+    }, [podMetaData,pods]);
 
     useEffect(() => {
         callBack(podTab === 'new');
+        setNewPods(newPodStats);
+        setOldPods(oldPodStats);
     }, [podTab]);
 
     return (
         <div className="pod-node-tab__wrapper flex left">
-            <div
-                className={`lh-1-4-33 no-decor pod-tab ${
-                    podTab === 'new' ? 'pod-tab__active' : ''
-                } border-right flex left column pl-16 pr-16 pointer `}
-                onClick={(e) => selectPodTab('new')}
-                data-testid="all-pods-new"
-            >
-                <div className="fs-14 fw-6 pt-12 ">New Pods ({newPods.length})</div>
-                <div className="flex left fs-12 cn-9 pb-12">
-                    <React.Fragment>
-                        {selectedHealthyNewNodeCount > 0 ? (
-                            <div className="pl-16"> {selectedHealthyNewNodeCount} healthy</div>
-                        ) : (
-                            ''
-                        )}
+            <TabSection podTab={podTab} selectPodTab={selectPodTab} podStatus={newPods} isNew={true} />
+            <TabSection podTab={podTab} selectPodTab={selectPodTab} podStatus={oldPods} isNew={false} />
+        </div>
+    )
+}
 
-                        {/* {<span className="bullet mr-4 ml-4"></span>} */}
-                        {/* <span data-testid={`new-pod-status`}> 8 running • 3 failed • 5 pending • 4 succeeded </span> */}
-                    </React.Fragment>
-                </div>
+function TabSection({
```

Comment:
Isolate this function as an individual component

Reply:
done